cairo-contracts/packages/core/src/commitment/types.cairo

```diff
@@ -133,25 +133,24 @@ pub impl StateProofZero of Zero<StateProof> {
 
 #[derive(Clone, Debug, Drop, PartialEq, Serde, starknet::Store)]
 pub struct StateRoot {
-    pub root: ByteArray // TODO: Determine the correct type (ByteArray or Array<u8>) once implemented membership proof verification.
-}
-
-pub impl ByteArrayIntoRoot of Into<ByteArray, StateRoot> {
-    fn into(self: ByteArray) -> StateRoot {
-        StateRoot { root: self }
-    }
+    // [u32; 8] is used over Array<u8> as it doesn't require manual
+    // implementation of `starknet::Store` and for Tendermint clients
+    //
+    // In addition, Array<u8> would require a conversion to [u32; 8] when
+    // verifying membership, since Cairo's sha256 hasher output [u32; 8]
```

Comment:
```suggestion
    // verifying membership, since Cairo's sha256 hasher output is [u32; 8]
    // where each u32 represents the hash bytes in little endian format.
```

Reply:
Done in 14a2e0b6605272f70210e1c8fe8fafabf2003e92